internal/cephfs/store/volumeoptions.go

```diff
@@ -235,6 +235,8 @@ func NewVolumeOptions(
 	opts.Monitors = strings.Join(clusterData.Monitors, ",")
 	opts.SubvolumeGroup = clusterData.CephFS.SubvolumeGroup
 	opts.Owner = k8s.GetOwner(volOptions)
+	// default BackingSnapshot to true only for ReadOnly volume requests
+	opts.BackingSnapshot = IsVolumeCreateRO(req.VolumeCapabilities)
```

Comment:
The datasource needs to be a snapshot for this to work.

Reply:
Here we are just specifying the value for backing snapshot based on the access mode of volume.
The data source should be handled by user, while creating the yaml of volume. right?

Reply:
> Here we are just specifying the value for backing snapshot based on the access mode of volume. The data source should be handled by user, while creating the yaml of volume. right?

This will cause a error when data source is pvc.

We should default backingSnapshot option to true if the datasource is a snapshot and request is for a RO volume.

Reply:
Thanks Rakshith, updated.